src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -116,20 +115,37 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
         ChatOptions? options = null,
         bool? useNativeJsonSchema = null,
         CancellationToken cancellationToken = default)
-        where T : class
     {
         _ = Throw.IfNull(chatClient);
         _ = Throw.IfNull(chatMessages);
         _ = Throw.IfNull(serializerOptions);
 
         serializerOptions.MakeReadOnly();
 
-        var schemaNode = AIJsonUtilities.CreateJsonSchema(
+        var schemaElement = AIJsonUtilities.CreateJsonSchema(
             type: typeof(T),
             serializerOptions: serializerOptions,
             inferenceOptions: _inferenceOptions);
 
-        var schema = JsonSerializer.Serialize(schemaNode, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonElement)));
+        var isWrappedInObject = false;
+
+        if (!SchemaRepresentsObject(schemaElement))
+        {
+            // We wrap regardless of native structured output, since it also applies to Azure Inference
```

Comment:
I don't understand this comment. Why is a specific service relevant here?

Reply:
The comment comes from @kzu's code.

I understand it to mean that among the real services we care about, which includes Azure Inference as an example, it's common for native structured output only to allow "object"-type JSON schemas. So to make this safe to use with non-object-type `T` values, we use the schema wrapping technique even if you're using native structured output.

Reply:
Marking as resolved so I can merge. @stephentoub if there are any concerns, let me know and I'll go back to it.

Reply:
Thanks. My concern is it looks like this is somehow an accomodation _specifically_ for an Azure service. It sounds like it's more of a case where this is a necessary accomodation for a whole class of services, in which case I'd rather it either not call out one specifically, or make it clear this is just an example of one.

Reply:
Makes sense. I did update the comment along those lines: https://github.com/dotnet/extensions/pull/5560/files#diff-e069bbb2b8d33d886c9d0545784be18532ba6d51716563a084499a921a0659bcR140